### PR TITLE
fix(meetings): fix to avoid llm connection update when its not enabled

### PR DESCRIPTION
**Pull request by Kesari3008**
# COMPLETES 

## This pull request addresses
In scenarios where LLM is not enabled before Webex initialization, we are sending a POST request to update the LLM connection during meeting cleanup.

## by making the following changes
Added a check in the cleanup function to check if meeting.config.enableAutomaticLLM  is set to true. If not, we do not invoke updateLLMConnection().

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested
Unchecked EnableLLM in Kitchen Sink App before webex initialization and joined the meeting. Triggered meeting.endMeetingForAll() from the console and verified that the registration request to the data channel url is not sent out post the end API call.
[Network_logs_with_fix.txt](https://github.com/webex/webex-js-sdk/files/14741974/Network_logs_with_fix.txt)

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/plugin-meetings/src/meeting/util.ts [hunk 1]
```diff
@@ -202,7 +202,11 @@ const MeetingUtil = {
         meeting.reconnectionManager.cleanUp();
       })
       .then(() => meeting.stopKeepAlive())
-      .then(() => meeting.updateLLMConnection());
+      .then(() => {
+        if (meeting.config.enableAutomaticLLM) {
```
Comment (mkesavan13) [inline packages/@webex/plugin-meetings/src/meeting/util.ts:206]:
```suggestion
        if (meeting?.config?.enableAutomaticLLM) {
```

**Review by mkesavan13 — APPROVED** [review]
Looks good to me.

**Review by Shreyas281299 — APPROVED** [review]
Approved. Please add the tested scenarios. Otherwise looks fine.

File: packages/@webex/plugin-meetings/src/meeting/util.ts (hunk 1 above)
Comment (sreenara) [reply packages/@webex/plugin-meetings/src/meeting/util.ts:206]: Can meeting and meeting.config be undefined here?

Comment (Kesari3008) [reply packages/@webex/plugin-meetings/src/meeting/util.ts]: added for the config as for meeting it wasn;t required

Comment (Kesari3008) [reply packages/@webex/plugin-meetings/src/meeting/util.ts]: meeting is not undefined but config can be

File: packages/@webex/plugin-meetings/src/meeting/util.ts [hunk 2]
```diff
@@ -202,7 +202,11 @@ const MeetingUtil = {
         meeting.reconnectionManager.cleanUp();
       })
       .then(() => meeting.stopKeepAlive())
-      .then(() => meeting.updateLLMConnection());
+      .then(() => {
+        if (meeting.config?.enableAutomaticLLM) {
```
Comment (sreenara) [inline packages/@webex/plugin-meetings/src/meeting/util.ts:206]: There needs to be one more UT with config object undefined.

**Review by sreenara — CHANGES_REQUESTED** [review]
Please check the comment

File: packages/@webex/plugin-meetings/src/meeting/util.ts (hunk 2 above)
Comment (Kesari3008) [reply packages/@webex/plugin-meetings/src/meeting/util.ts:206]: Added

**Review by sreenara — APPROVED** [review]

**Pull request merged**